### DISCUSSION TITLE
The connection to S7 1500 gets interrupted while the connection to S7 1200 does not.

### DISCUSSION THREAD

**Discussion by lyworry** [eclipse-milo/milo#1195] [General]
**Describe the bug**
The program developed using Milo to connect to the Siemens S7 1500 initially establishes communication successfully, but after a while, the connection is interrupted. Upon analyzing the network packets, it can be observed that the server side is not responding in a timely manner, leading to a connection reset. However, when tested with UaExpert, the communication remains stable. Additionally, when connecting the Milo-developed program to the Siemens S7 1200, the communication remains uninterrupted without experiencing the aforementioned issue. I am unable to identify the root cause of this problem. Could it be due to excessive server-side resource pressure on the S7 1500? However, Siemens technical support informed me that if the S7 1500 experiences excessive pressure, it should return an exception indicating insufficient resources.

**Logs and Packet Captures**
![image](https://github.com/eclipse/milo/assets/32288518/3fb682a3-cb24-458d-84fc-4406bcc53e92)

**Additional context**

        <dependency>
            <groupId>org.eclipse.milo</groupId>
            <artifactId>sdk-server</artifactId>
            <version>0.6.8</version>
        </dependency>

        
        <dependency>
            <groupId>org.eclipse.milo</groupId>
            <artifactId>sdk-client</artifactId>
            <version>0.6.8</version>
        </dependency>

**Comment by kevinherron** [comment]
I'm not sure I can help you much with just a screenshot of a packet capture at the time the connection fails.

You will probably have to do a long term analysis of how your application communicates with the S7-1500, look for anything out of place like continual creation of sessions, subscriptions, or monitored items, and possibly continue working with Siemens support in case it's simply an issue with the server.

Milo is used for the OPC UA implementation at the company I work for and there are many connections to S7-1200 and S7-1500 across many customers but this is not an issue I have seen reported.

**Comment by vcatrici** [comment]
@lyworry did you get chance to analyse and/or root cause? I'm seeing the same behaviour with S7 1500 but only on a certain firmware version.
